USE rna_seq;

DROP TABLE IF EXISTS `file_queue`;
CREATE TABLE `file_queue` (
  `id` INT(4) NOT NULL AUTO_INCREMENT,
  `filename` VARCHAR(255) NOT NULL,
  `qc_source` VARCHAR(128),
  `qc_date` DATETIME NOT NULL, 
  `type` VARCHAR(64) NULL,
  `received_at` TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
  `processed_at` TIMESTAMP NULL,
  PRIMARY KEY (`id`),
  UNIQUE INDEX `file_name_UNIQUE` (`filename` ASC))
ENGINE = InnoDB
DEFAULT CHARACTER SET = utf8;

DROP TABLE IF EXISTS `samples`;
CREATE TABLE `samples` (
  `id` VARCHAR(64) NOT NULL,
  `study_id` VARCHAR(128) NOT NULL,
  `center` VARCHAR(32) NOT NULL,
  `dataset_id` VARCHAR(255) NOT NULL,
  `torid` VARCHAR(128) NULL DEFAULT NULL,
  `investigator_id` VARCHAR(128) NULL DEFAULT NULL,
  `family_id` VARCHAR(128) NULL DEFAULT NULL,
  `good_faith_approved` VARCHAR(8) NULL DEFAULT NULL,  
  `tissue_type` VARCHAR(128) NULL DEFAULT NULL,
  `arrival_date` DATETIME NOT NULL,
  PRIMARY KEY (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8;

DROP TABLE IF EXISTS `qc_attributes_mapper`;
CREATE TABLE `qc_attributes_mapper` (
  `id` int(5) NOT NULL AUTO_INCREMENT,
  `field_name` varchar(64) NOT NULL,
  `full_attribute` varchar(255) NOT NULL,
  `description` text,
  `source` varchar(64) DEFAULT NULL,
  `weight` int(1) NOT NULL DEFAULT '1',
  `mean` float DEFAULT NULL,
  `sd` float DEFAULT NULL,
  `min` float DEFAULT NULL,
  `pct25` float DEFAULT NULL,
  `median` float DEFAULT NULL,
  `pct75` float DEFAULT NULL,
  `max` float DEFAULT NULL,
  `updated_at` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
  PRIMARY KEY (`id`),
  UNIQUE KEY `extract_key_UNIQUE` (`field_name`),
  UNIQUE KEY `description_UNIQUE` (`full_attribute`)
) ENGINE=InnoDB  DEFAULT CHARSET=utf8;


INSERT INTO `qc_attributes_mapper` VALUES 
(1,'sample_id','Sample',NULL,NULL,1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(2,'mapping_rate','Mapping Rate','The proportion of all reads in the Bam which were Mapped, and not Secondary Alignments or Platform/Vendor QC Failing reads (\"Mapped Reads\")\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(3,'unique_rate_of_mapped','Unique Rate of Mapped','The proportion of \"Mapped Reads\" which were not marked as duplicates\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(4,'dupli_rate_mapped','Duplicate Rate of Mapped','The proportion of \"Mapped Reads\" which were marked as duplicates\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(5,'dupli_rate_no_globins','Duplicate Rate of Mapped, excluding Globins','This is similar to the \"Duplicate Rate of Mapped\" except that it only includes reads which did not align to HBA1, HBA2, HBB, or HBD','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(6,'base_mismatch','Base Mismatch','The total number of mismatched bases (as determined by the \"NM\" tag) of all \"Mapped Reads\" divided by the total aligned length of all \"Mapped Reads\"\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(7,'end_1_mapping_rate','End 1 Mapping Rate','The proportion of Paired reads which were marked as First in the pair out of all \"Mapped Reads\"\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(8,'end_2_mapping_rate','End 2 Mapping Rate','The proportion of Paired reads which were marked as Second in the pair out of all \"Mapped Reads\"\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(9,'end_1_mismatch_rate','End 1 Mismatch Rate','The proportion of mismatched bases (as determined by the \"NM\" tag) belonging to First  mates, divided by the total aligned length of all \"Mapped\" First mates\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(10,'end_2_mismatch_rate','End 2 Mismatch Rate','The proportion of mismatched bases (as determined by the \"NM\" tag) belonging to Second mates, divided by the total aligned length of all \"Mapped\" Second mates\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(11,'expression_profiling_efficiency','Expression Profiling Efficiency','The proportion of \"Exonic Reads\" (see \"Exonic Rate\") out of all reads which were not Secondary Alignments or Platform/Vendor QC Failing reads\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(12,'high_quality_rate','High Quality Rate','The proportion of properly paired reads with less than 6 mismatched bases and a perfect mapping quality out of all \"Mapped Reads\"\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(13,'exonic_rate','Exonic Rate','The proportion of \"Mapped Reads\" for which all aligned segments unambiguously aligned to exons of the same gene\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(14,'intronic_rate','Intronic Rate','The proportion of \"Mapped Reads\" for which all aligned segments unambiguously aligned to the same gene, but none of which intersected any exons of the gene\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(15,'intergenic_rate','Intergenic Rate','The proportion of \"Mapped Reads\" for which none of the aligned segments intersected any genes\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),(16,'intragenic_rate','Intragenic Rate','The sum of \"Exonic Rate\" and \"Intronic Rate\"','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(17,'ambiguous_alignment_rate','Ambiguous Alignment Rate','The proportion of \"Mapped Reads\" where the aligned segments unambiguously aligned to exons of more than one gene','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(18,'high_quality_exonic_rate','High Quality Exonic Rate','The proportion of \"Exonic Reads\" (as defined in \"Exonic Rate\") out of \"High Quality Reads\" only (as defined in \"High Quality Rate\")\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(19,'high_quality_intronic_rate','High Quality Intronic Rate','The proportion of \"Intronic Reads\" (as defined in \"Intronic Rate\") out of \"High Quality Reads\" only (as defined in \"High Quality Rate\")\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(20,'high_quality_intergenic_rate','High Quality Intergenic Rate','The proportion of \"Intergenic Reads\" (as defined in \"Intergenic Rate\") out of \"High Quality Reads\" only (as defined in \"High Quality Rate\")\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(21,'high_quality_intragenic_rate','High Quality Intragenic Rate','The proportion of \"Intragenic Reads\" (as defined in \"Intragenic Rate\") out of \"High Quality Reads\" only (as defined in \"High Quality Rate\")\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(22,'high_quality_ambiguous_alignment_rate','High Quality Ambiguous Alignment Rate','The proportion of \"Ambiguous Alignments\" (as defined in \"Ambiguous Alignment Rate\") out of \"High Quality Reads\" only (as defined in \"High Quality Rate\")\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(23,'discard_rate','Discard Rate','The proportion of \"Mapped Reads\" which were discarded and not checked against the reference annotation. \"Exonic Rate\", \"Intronic Rate\", \"Intergenic Rate\", \"Ambiguous Alignment Rate\" and \"Discard Rate\" will sum to 1.\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(24,'rrna_rate','rRNA Rate','The proportion of \"Mapped Reads\" which at least partially intersected with an annotated rRNA gene\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(25,'chimeric_alignment_rate','Chimeric Alignment Rate','The proportion of \"Mapped Reads\" which are \"Chimeric Reads\"\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(26,'end_1_sense_rate','End 1 Sense Rate','The proportion of First Mate reads which intersected with a Sense Strand feature out of all First Mate reads which intersected with any features (End 1 Sense / (End 1 Sense + End 1 Antisense))\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(27,'end_2_sense_rate','End 2 Sense Rate','The proportion of Second Mate reads which intersected with a Sense Strand feature out of all Second Mate reads which intersected with any features (End 2 Sense / (End 2 Sense + End 2 Antisense))\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(28,'avg_splits_per_read','Avg. Splits per Read','The average number of gaps or deletions present in \"Mapped Reads\". This is generally not an important metric, but may indicate aligner errors if the value is too high.\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(29,'alter_alignments','Alternative Alignments','The number of secondary alignments (0x100 flag set)\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(30,'chimeric_reads','Chimeric Reads','Number of chimeric reads, as defined as mates mapping more than 2Mb apart\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(31,'duplicate_reads','Duplicate Reads','Number of \"Unique Mapping, Vendor QC Passed Reads\" marked as duplicate\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(32,'end_1_antisense','End 1 Antisense','The number of First Mate reads which intersected with an Antisense Strand feature\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(33,'end_2_antisense','End 2 Antisense','The number of Second Mate reads which intersected with an Antisense Strand feature\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(34,'end_1_bases','End 1 Bases','The total aligned length of all paired First Mate \"Mapped Reads\"\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(35,'end_2_bases','End 2 Bases','The total aligned length of all paired Second Mate \"Mapped Reads\"\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(36,'end_1_mapped_reads','End 1 Mapped Reads','The number of paired \"Mapped Reads\" flagged as First\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(37,'end_2_mapped_reads','End 2 Mapped Reads','The number of paired \"Mapped Reads\" flagged as Second\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(38,'end_1_mismatches','End 1 Mismatches','The number of total mismatches (according to \'NM\' tag) in First Mate \"Unique Mapping, Vendor QC Passed Reads\"\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(39,'end_2_mismatches','End 2 Mismatches','The number of total mismatches (according to \'NM\' tag) in Second Mate \"Unique Mapping, Vendor QC Passed Reads\"\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(40,'end_1_sense','End 1 Sense','The number of First Mate reads which intersected with a Sense Strand feature\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(41,'end_2_sense','End 2 Sense','The number of Second Mate reads which intersected with a Sense Strand feature\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(42,'exonic_reads','Exonic Reads','The number of \"Mapped Reads\" for which all aligned segments unambiguously aligned to exons of the same gene\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(43,'failed_vendor_qc','Failed Vendor QC','The number of alignments with the 0x200 flag set (excluding secondary alignments)\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(44,'high_quality_reads','High Quality Reads','The number of \"Mapped Reads\" that were properly paired with less than 6 mismatched bases and a perfect mapping quality\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(45,'intergenic_reads','Intergenic Reads','The number of \"Mapped Reads\" for which none of the aligned segments intersected any genes\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(46,'intragenic_reads','Intragenic Reads','The sum of \"Exonic Reads\" and \"Intronic Reads\"\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(47,'ambiguous_reads','Ambiguous Reads','The number of \"Mapped Reads\" where the aligned segments unambiguously aligned to exons of more than one gene\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(48,'intronic_reads','Intronic Reads','The number of \"Mapped Reads\" for which all aligned segments unambiguously aligned to the same gene, but none of which intersected any exons of the gene\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(49,'low_mapping_quality','Low Mapping Quality','The number of alignments with mapping quality < 255 (excluding \"Alternative Alignments\" and \"Failed Vendor QC\" alignments)\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(50,'low_quality_reads','Low Quality Reads','The number of \"Mapped Reads\" that were not \"High Quality Reads\"\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(51,'mapped_duplicate_reads','Mapped Duplicate Reads','The number of \"Mapped Reads\" that were marked as duplicates\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(52,'mapped_reads','Mapped Reads','The number of all reads in the bam which were Mapped, and not Secondary Alignments or Platform/Vendor QC Failing reads\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(53,'mapped_unique_reads','Mapped Unique Reads','The number of \"Mapped Reads\" that were not marked as duplicates\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(54,'mismatched_bases','Mismatched Bases','The sum of \"End 1 Mismatches\" and \"End 2 Mismatches\"\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(55,'non_globin_reads','Non-Globin Reads','The number of mapped reads that did not align to HBA1, HBA2, HBB, or HBD\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(56,'non_globin_duplicate_reads','Non-Globin Duplicate Reads','The number of \"Non-Globin Reads\" marked as duplicates\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(57,'reads_excluded_from_exon_counts','Reads excluded from exon counts',NULL,'RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(58,'reads_used_for_intron_exon_counts','Reads used for Intron/Exon counts','\"Mapped Reads\" that aligned to a known reference chromosome (should be equal to \"Mapped Reads\")\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(59,'rrna_reads','rRNA Reads','The number of \"Mapped Reads\" which at least partially intersected with an annotated rRNA gene\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(60,'total_bases','Total Bases','The total aligned length of all \"Mapped Reads\"\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(61,'total_mapped_pairs','Total Mapped Pairs','Calculated as number of \"Mapped Reads\" marked as First Mate, paired with mate mapped\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(62,'total_reads','Total Reads','The number of total alignments in the BAM file\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(63,'unique_mapping_vendor_qc_passed_reads','Unique Mapping, Vendor QC Passed Reads','The number of alignments excluding \"Alternative Alignments\" and \"Failed Vendor QC\" alignments\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(64,'unpaired_reads','Unpaired Reads','The number of Unique Mapping, Vendor QC Passed Reads that are unpaired\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(65,'read_length','Read Length','The longest aligned length observed in any read\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(66,'genes_detected','Genes Detected','The number of genes which had at least 5 unambiguous reads\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(67,'estimated_library_complexity','Estimated Library Complexity','An estimation of the number of unique cDNA fragments present in the library. This computation follows the same formula as Picard EstimateLibraryComplexity\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(68,'genes_used_in_3_bias','Genes used in 3\' bias','The number of genes used to calculate the 3\' bias metrics\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(69,'mean_3_bias','Mean 3\' bias','Mean 3\' bias based on the total coverage in 100 bp windows on both the 3\' and 5\' ends of a gene. The windows are both offset 150 bases into the gene. This computation is only performed on genes at least 600bp long and with at least 5 unambiguous reads. A gene with even coverage in both it\'s 3\' and 5\' windows would have a bias of 0.5; bias near 1 or 0 may indicate degredation\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(70,'median_3_bias','Median 3\' bias','Median 3\' bias based on the total coverage in 100 bp windows on both the 3\' and 5\' ends of a gene. The windows are both offset 150 bases into the gene. This computation is only performed on genes at least 600bp long and with at least 5 unambiguous reads. A gene with even coverage in both it\'s 3\' and 5\' windows would have a bias of 0.5; bias near 1 or 0 may indicate degredation\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(71,'3_bias_std','3\' bias Std','Standard deviation of 3\' bias based on the total coverage in 100 bp windows on both the 3\' and 5\' ends of a gene. The windows are both offset 150 bases into the gene. This computation is only performed on genes at least 600bp long and with at least 5 unambiguous reads. A gene with even coverage in both it\'s 3\' and 5\' windows would have a bias of 0.5; bias near 1 or 0 may indicate degredation\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(72,'3_bias_mad_std','3\' bias MAD_Std','Median absolute deviation based estimate of standard deviation of 3\' bias based on the total coverage in 100 bp windows on both the 3\' and 5\' ends of a gene. The windows are both offset 150 bases into the gene. This computation is only performed on genes at least 600bp long and with at least 5 unambiguous reads. A gene with even coverage in both it\'s 3\' and 5\' windows would have a bias of 0.5; bias near 1 or 0 may indicate degredation\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(73,'3_bias_25th_percentile','3\' Bias, 25th Percentile','25th quantile of 3\' bias based on the total coverage in 100 bp windows on both the 3\' and 5\' ends of a gene. The windows are both offset 150 bases into the gene. This computation is only performed on genes at least 600bp long and with at least 5 unambiguous reads. A gene with even coverage in both it\'s 3\' and 5\' windows would have a bias of 0.5; bias near 1 or 0 may indicate degredation\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(74,'3_bias_75th_percentile','3\' Bias, 75th Percentile','75th quantile of 3\' bias based on the total coverage in 100 bp windows on both the 3\' and 5\' ends of a gene. The windows are both offset 150 bases into the gene. This computation is only performed on genes at least 600bp long and with at least 5 unambiguous reads. A gene with even coverage in both it\'s 3\' and 5\' windows would have a bias of 0.5; bias near 1 or 0 may indicate degredation\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(75,'average_fragment_length','Average Fragment Length','Mean insert sizes observed in \"High Quality\" read pairs (only the first 1,000,000 \"High Quality\" pairs, where both mates map to the same Bed interval are used)\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(76,'fragment_length_median','Fragment Length Median','Median insert sizes observed in \"High Quality\" read pairs (only the first 1,000,000 \"High Quality\" pairs, where both mates map to the same Bed interval are used)\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(77,'fragment_length_std','Fragment Length Std','Std. deviation of insert sizes observed in \"High Quality\" read pairs (only the first 1,000,000 \"High Quality\" pairs, where both mates map to the same Bed interval are used)\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(78,'fragment_length_mad_std','Fragment Length MAD_Std','Median absolute deviation based estimate of standard deviation (using scale factor = 1.4826) of insert sizes observed in \"High Quality\" read pairs (only the first 1,000,000 \"High Quality\" pairs, where both mates map to the same Bed interval are used)\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(79,'median_of_avg_transcript_coverage','Median of Avg Transcript Coverage','Median of mean transcript coverage. Transcript coverage is computed by dropping the first and last 500bp of each gene and measuring the \"High Quality\" coverage over the remainder of the gene\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(80,'median_of_transcript_coverage_std','Median of Transcript Coverage Std','Median of standard deviation of transcript coverage. Transcript coverage is computed by dropping the first and last 500bp of each gene and measuring the \"High Quality\" coverage over the remainder of the gene\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(81,'median_of_transcript_coverage_cv','Median of Transcript Coverage CV','Median of coefficient of variation of transcript coverage. Transcript coverage is computed by dropping the first and last 500bp of each gene and measuring the \"High Quality\" coverage over the remainder of the gene\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(82,'median_exon_cv','Median Exon CV','Median coefficient of variation of exon coverage. Exon coverage is computed by dropping the first and last 500bp of each gene and measuring the \"High Quality\" coverage over the remainder of the exons. A lower value indicates more consistent coverage over exons.\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20'),
(83,'exon_cv_mad','Exon CV MAD','The Median Absolute Deviation over all coefficients of variation of exon coverage. Exon coverage is computed by dropping the first and last 500bp of each gene and measuring the \"High Quality\" coverage over the remainder of the exons\n','RNA-SeQC',1,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'2020-10-08 21:20:20');

--
-- Rank metrics for display
-- 
UPDATE `qc_attributes_mapper` SET `weight`='50' WHERE `id`='12';
UPDATE `qc_attributes_mapper` SET `weight`='48' WHERE `id`='2';
UPDATE `qc_attributes_mapper` SET `weight`='46' WHERE `id`='18';
UPDATE `qc_attributes_mapper` SET `weight`='44' WHERE `id`='19';
UPDATE `qc_attributes_mapper` SET `weight`='42' WHERE `id`='20';
UPDATE `qc_attributes_mapper` SET `weight`='38' WHERE `id`='70';
UPDATE `qc_attributes_mapper` SET `weight`='36' WHERE `id`='66';
UPDATE `qc_attributes_mapper` SET `weight`='30' WHERE `id`='4';
UPDATE `qc_attributes_mapper` SET `weight`='28' WHERE `id`='76';
UPDATE `qc_attributes_mapper` SET `weight`='20' WHERE `id`='24';

DROP TABLE IF EXISTS `qc_metrics`;
CREATE TABLE `qc_metrics` (
  `sample_id` VARCHAR(64) NOT NULL,
  `file_queue_id` INT(4) NULL DEFAULT NULL,
  `mapping_rate` FLOAT NULL DEFAULT NULL,
  `unique_rate_of_mapped` FLOAT NULL DEFAULT NULL,
  `dupli_rate_mapped` FLOAT NULL DEFAULT NULL,
  `dupli_rate_no_globins` FLOAT NULL DEFAULT NULL,
  `base_mismatch` FLOAT NULL DEFAULT NULL,
  `end_1_mapping_rate` FLOAT NULL DEFAULT NULL,
  `end_2_mapping_rate` FLOAT NULL DEFAULT NULL,
  `end_1_mismatch_rate` FLOAT NULL DEFAULT NULL,
  `end_2_mismatch_rate` FLOAT NULL DEFAULT NULL,
  `expression_profiling_efficiency` FLOAT NULL DEFAULT NULL,
  `high_quality_rate` FLOAT NULL DEFAULT NULL,
  `exonic_rate` FLOAT NULL DEFAULT NULL,
  `intronic_rate` FLOAT NULL DEFAULT NULL,
  `intergenic_rate` FLOAT NULL DEFAULT NULL,
  `intragenic_rate` FLOAT NULL DEFAULT NULL,
  `ambiguous_alignment_rate` FLOAT NULL DEFAULT NULL,
  `high_quality_exonic_rate` FLOAT NULL DEFAULT NULL,
  `high_quality_intronic_rate` FLOAT NULL DEFAULT NULL,
  `high_quality_intergenic_rate` FLOAT NULL DEFAULT NULL,
  `high_quality_intragenic_rate` FLOAT NULL DEFAULT NULL,
  `high_quality_ambiguous_alignment_rate` FLOAT NULL DEFAULT NULL,
  `discard_rate` FLOAT NULL DEFAULT NULL,
  `rrna_rate` FLOAT NULL DEFAULT NULL,
  `chimeric_alignment_rate` FLOAT NULL DEFAULT NULL,
  `end_1_sense_rate` FLOAT NULL DEFAULT NULL,
  `end_2_sense_rate` FLOAT NULL DEFAULT NULL,
  `avg_splits_per_read` FLOAT NULL DEFAULT NULL,
  `alter_alignments` FLOAT NULL DEFAULT NULL,
  `chimeric_reads` FLOAT NULL DEFAULT NULL,
  `duplicate_reads` FLOAT NULL DEFAULT NULL,
  `end_1_antisense` FLOAT NULL DEFAULT NULL,
  `end_2_antisense` FLOAT NULL DEFAULT NULL,
  `end_1_bases` FLOAT NULL DEFAULT NULL,
  `end_2_bases` FLOAT NULL DEFAULT NULL,
  `end_1_mapped_reads` FLOAT NULL DEFAULT NULL,
  `end_2_mapped_reads` FLOAT NULL DEFAULT NULL,
  `end_1_mismatches` FLOAT NULL DEFAULT NULL,
  `end_2_mismatches` FLOAT NULL DEFAULT NULL,
  `end_1_sense` FLOAT NULL DEFAULT NULL,
  `end_2_sense` FLOAT NULL DEFAULT NULL,
  `exonic_reads` FLOAT NULL DEFAULT NULL,
  `failed_vendor_qc` FLOAT NULL DEFAULT NULL,
  `high_quality_reads` FLOAT NULL DEFAULT NULL,
  `intergenic_reads` FLOAT NULL DEFAULT NULL,
  `intragenic_reads` FLOAT NULL DEFAULT NULL,
  `ambiguous_reads` FLOAT NULL DEFAULT NULL,
  `intronic_reads` FLOAT NULL DEFAULT NULL,
  `low_mapping_quality` FLOAT NULL DEFAULT NULL,
  `low_quality_reads` FLOAT NULL DEFAULT NULL,
  `mapped_duplicate_reads` FLOAT NULL DEFAULT NULL,
  `mapped_reads` FLOAT NULL DEFAULT NULL,
  `mapped_unique_reads` FLOAT NULL DEFAULT NULL,
  `mismatched_bases` FLOAT NULL DEFAULT NULL,
  `non_globin_reads` FLOAT NULL DEFAULT NULL,
  `non_globin_duplicate_reads` FLOAT NULL DEFAULT NULL,
  `reads_excluded_from_exon_counts` FLOAT NULL DEFAULT NULL,
  `reads_used_for_intron_exon_counts` FLOAT NULL DEFAULT NULL,
  `rrna_reads` FLOAT NULL DEFAULT NULL,
  `total_bases` FLOAT NULL DEFAULT NULL,
  `total_mapped_pairs` FLOAT NULL DEFAULT NULL,
  `total_reads` FLOAT NULL DEFAULT NULL,
  `unique_mapping_vendor_qc_passed_reads` FLOAT NULL DEFAULT NULL,
  `unpaired_reads` FLOAT NULL DEFAULT NULL,
  `read_length` FLOAT NULL DEFAULT NULL,
  `genes_detected` FLOAT NULL DEFAULT NULL,
  `estimated_library_complexity` FLOAT NULL DEFAULT NULL,
  `genes_used_in_3_bias` FLOAT NULL DEFAULT NULL,
  `mean_3_bias` FLOAT NULL DEFAULT NULL,
  `median_3_bias` FLOAT NULL DEFAULT NULL,
  `3_bias_std` FLOAT NULL DEFAULT NULL,
  `3_bias_mad_std` FLOAT NULL DEFAULT NULL,
  `3_bias_25th_percentile` FLOAT NULL DEFAULT NULL,
  `3_bias_75th_percentile` FLOAT NULL DEFAULT NULL,
  `average_fragment_length` FLOAT NULL DEFAULT NULL,
  `fragment_length_median` FLOAT NULL DEFAULT NULL,
  `fragment_length_std` FLOAT NULL DEFAULT NULL,
  `fragment_length_mad_std` FLOAT NULL DEFAULT NULL,
  `median_of_avg_transcript_coverage` FLOAT NULL DEFAULT NULL,
  `median_of_transcript_coverage_std` FLOAT NULL DEFAULT NULL,
  `median_of_transcript_coverage_cv` FLOAT NULL DEFAULT NULL,
  `median_exon_cv` FLOAT NULL DEFAULT NULL,
  `exon_cv_mad` FLOAT NULL DEFAULT NULL,
  PRIMARY KEY (`sample_id`))
ENGINE = InnoDB
DEFAULT CHARACTER SET = utf8;

DROP TABLE IF EXISTS `studies`;
CREATE TABLE `studies` (
  `id` int(3) NOT NULL AUTO_INCREMENT,
  `study` VARCHAR(64) NOT NULL,
  `pi` VARCHAR(64) NULL,
  `datatype` VARCHAR(32) NOT NULL,
  `center` VARCHAR(64) NOT NULL,
  `samplereceived` int(8) NULL,
  `updated_at` TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
  PRIMARY KEY (`id`),
  INDEX `idx_study_datatype` (`study` ASC, `datatype` ASC)
) ENGINE=InnoDB DEFAULT CHARSET=utf8;

