USE rna_seq;

DROP TABLE IF EXISTS `file_queue`;
CREATE TABLE `file_queue` (
  `id` INT(4) NOT NULL AUTO_INCREMENT,
  `filename` VARCHAR(255) NOT NULL,
  `qc_source` VARCHAR(128),
  `qc_date` DATETIME NOT NULL, 
  `type` VARCHAR(64) NULL,
  `received_at` TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
  `processed_at` TIMESTAMP NULL,
  PRIMARY KEY (`id`),
  UNIQUE INDEX `file_name_UNIQUE` (`filename` ASC))
ENGINE = InnoDB
DEFAULT CHARACTER SET = utf8;

DROP TABLE IF EXISTS `samples`;
CREATE TABLE `samples` (
  `id` VARCHAR(64) NOT NULL,
  `study_id` VARCHAR(128) NOT NULL,
  `dataset_id` VARCHAR(255) NOT NULL,
  `torid` VARCHAR(128) NULL DEFAULT NULL,  
  `investigator_id` VARCHAR(128) NULL DEFAULT NULL,  
  `family_id` VARCHAR(128) NULL DEFAULT NULL,  
  `good_faith_approved` VARCHAR(8) NULL DEFAULT NULL,  
  `tissue_type` VARCHAR(128) NULL DEFAULT NULL,
  `arrival_date` DATETIME NOT NULL,
  PRIMARY KEY (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8;

DROP TABLE IF EXISTS `qc_attributes_mapper`;
CREATE TABLE `qc_attributes_mapper` (
  `id` INT(5) NOT NULL AUTO_INCREMENT,
  `field_name` VARCHAR(64) NOT NULL,
  `full_attribute` VARCHAR(255) NOT NULL,
  `weight` INT(1) NOT NULL DEFAULT 1,
  `mean` FLOAT NULL DEFAULT NULL,
  `sd` FLOAT NULL DEFAULT NULL,
  `min` FLOAT NULL DEFAULT NULL,
  `pct25` FLOAT NULL DEFAULT NULL,
  `median` FLOAT NULL DEFAULT NULL,
  `pct75` FLOAT NULL DEFAULT NULL,
  `max` FLOAT NULL DEFAULT NULL,
  `updated_at` TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
  PRIMARY KEY (`id`),
  UNIQUE INDEX `extract_key_UNIQUE` (`field_name` ASC),
  UNIQUE INDEX `description_UNIQUE` (`full_attribute` ASC))
ENGINE = InnoDB
DEFAULT CHARACTER SET = utf8;

INSERT INTO `qc_attributes_mapper` (`id`, `full_attribute`, `field_name`) VALUES
(1, "Sample", "sample_id"),
(2, "Mapping Rate", "mapping_rate"),
(3, "Unique Rate of Mapped", "unique_rate_of_mapped"),
(4, "Duplicate Rate of Mapped", "dupli_rate_mapped"),
(5, "Duplicate Rate of Mapped, excluding Globins", "dupli_rate_no_globins"),
(6, "Base Mismatch", "base_mismatch"),
(7, "End 1 Mapping Rate", "end_1_mapping_rate"),
(8, "End 2 Mapping Rate", "end_2_mapping_rate"),
(9, "End 1 Mismatch Rate", "end_1_mismatch_rate"),
(10, "End 2 Mismatch Rate", "end_2_mismatch_rate"),
(11, "Expression Profiling Efficiency", "expression_profiling_efficiency"),
(12, "High Quality Rate", "high_quality_rate"),
(13, "Exonic Rate", "exonic_rate"),
(14, "Intronic Rate", "intronic_rate"),
(15, "Intergenic Rate", "intergenic_rate"),
(16, "Intragenic Rate", "intragenic_rate"),
(17, "Ambiguous Alignment Rate", "ambiguous_alignment_rate"),
(18, "High Quality Exonic Rate", "high_quality_exonic_rate"),
(19, "High Quality Intronic Rate", "high_quality_intronic_rate"),
(20, "High Quality Intergenic Rate", "high_quality_intergenic_rate"),
(21, "High Quality Intragenic Rate", "high_quality_intragenic_rate"),
(22, "High Quality Ambiguous Alignment Rate", "high_quality_ambiguous_alignment_rate"),
(23, "Discard Rate", "discard_rate"),
(24, "rRNA Rate", "rrna_rate"),
(25, "Chimeric Alignment Rate", "chimeric_alignment_rate"),
(26, "End 1 Sense Rate", "end_1_sense_rate"),
(27, "End 2 Sense Rate", "end_2_sense_rate"),
(28, "Avg. Splits per Read", "avg_splits_per_read"),
(29, "Alternative Alignments", "alter_alignments"),
(30, "Chimeric Reads", "chimeric_reads"),
(31, "Duplicate Reads", "duplicate_reads"),
(32, "End 1 Antisense", "end_1_antisense"),
(33, "End 2 Antisense", "end_2_antisense"),
(34, "End 1 Bases", "end_1_bases"),
(35, "End 2 Bases", "end_2_bases"),
(36, "End 1 Mapped Reads", "end_1_mapped_reads"),
(37, "End 2 Mapped Reads", "end_2_mapped_reads"),
(38, "End 1 Mismatches", "end_1_mismatches"),
(39, "End 2 Mismatches", "end_2_mismatches"),
(40, "End 1 Sense", "end_1_sense"),
(41, "End 2 Sense", "end_2_sense"),
(42, "Exonic Reads", "exonic_reads"),
(43, "Failed Vendor QC", "failed_vendor_qc"),
(44, "High Quality Reads", "high_quality_reads"),
(45, "Intergenic Reads", "intergenic_reads"),
(46, "Intragenic Reads", "intragenic_reads"),
(47, "Ambiguous Reads", "ambiguous_reads"),
(48, "Intronic Reads", "intronic_reads"),
(49, "Low Mapping Quality", "low_mapping_quality"),
(50, "Low Quality Reads", "low_quality_reads"),
(51, "Mapped Duplicate Reads", "mapped_duplicate_reads"),
(52, "Mapped Reads", "mapped_reads"),
(53, "Mapped Unique Reads", "mapped_unique_reads"),
(54, "Mismatched Bases", "mismatched_bases"),
(55, "Non-Globin Reads", "non_globin_reads"),
(56, "Non-Globin Duplicate Reads", "non_globin_duplicate_reads"),
(57, "Reads excluded from exon counts", "reads_excluded_from_exon_counts"),
(58, "Reads used for Intron\/Exon counts", "reads_used_for_intron_exon_counts"),
(59, "rRNA Reads", "rrna_reads"),
(60, "Total Bases", "total_bases"),
(61, "Total Mapped Pairs", "total_mapped_pairs"),
(62, "Total Reads", "total_reads"),
(63, "Unique Mapping, Vendor QC Passed Reads", "unique_mapping_vendor_qc_passed_reads"),
(64, "Unpaired Reads", "unpaired_reads"),
(65, "Read Length", "read_length"),
(66, "Genes Detected", "genes_detected"),
(67, "Estimated Library Complexity", "estimated_library_complexity"),
(68, "Genes used in 3' bias", "genes_used_in_3_bias"),
(69, "Mean 3' bias", "mean_3_bias"),
(70, "Median 3' bias", "median_3_bias"),
(71, "3' bias Std", "3_bias_std"),
(72, "3' bias MAD_Std", "3_bias_mad_std"),
(73, "3' Bias, 25th Percentile", "3_bias_25th_percentile"),
(74, "3' Bias, 75th Percentile", "3_bias_75th_percentile"),
(75, "Average Fragment Length", "average_fragment_length"),
(76, "Fragment Length Median", "fragment_length_median"),
(77, "Fragment Length Std", "fragment_length_std"),
(78, "Fragment Length MAD_Std", "fragment_length_mad_std"),
(79, "Median of Avg Transcript Coverage", "median_of_avg_transcript_coverage"),
(80, "Median of Transcript Coverage Std", "median_of_transcript_coverage_std"),
(81, "Median of Transcript Coverage CV", "median_of_transcript_coverage_cv"),
(82, "Median Exon CV", "median_exon_cv"),
(83, "Exon CV MAD", "exon_cv_mad"),
(84, "Note", "note"),
(85, "Chimeric Pairs", "chimeric_pairs"),
(86, "Num. Gaps", "num_gaps"),
(87, "5' Norm", "5_norm"),
(88, "3' Norm", "3_norm"),
(89, "Mean Per Base Cov.", "mean_per_base_cov"),
(90, "Fragment Length StdDev", "fragment_length_stdDev"),
(91, "Estimated Library Size", "estimated_library_size"),
(92, "Mapped", "mapped"),
(93, "Total Purity Filtered Reads Sequenced", "total_purity_filtered_reads_sequenced"),
(94, "rRNA", "rrna"),
(95, "Failed Vendor QC Check", "failed_vendor_qc_check"),
(96, "Mean CV","mean_cv"),
(97, "Transcripts Detected", "transcripts_detected"),
(98, "Mapped Pairs", "mapped_paris"),
(99, "Cumul. Gap Length","cumul_gap_length"),
(100, "Gap %", "gap_pct"),
(101, "Mapped Unique Rate of Total", "mapped_unique_rate_of_total"),
(102, "Mapped Unique", "mapped_unique"),
(103, "Alternative Aligments", "alter_aligments"),
(104, "Fragment Length Mean", "fragment_length_mean"),
(105, "Split Reads", "split_reads"),
(106, "Base Mismatch Rate", "base_mismatch_rate"),
(107, "End 1 % Sense", "end_1_pct_sense"),
(108, "No. Covered 5'", "num_covered_5"),
(109, "Duplication Rate of Mapped", "duplication_rate_of_mapped"),
(110, "End 2 % Sense", "end_2_pct_sense");


DROP TABLE IF EXISTS `qc_metrics`;
CREATE TABLE `qc_metrics` (
  `sample_id` VARCHAR(64) NOT NULL,
  `file_queue_id` INT(4) NULL DEFAULT NULL,
  `mapping_rate` FLOAT NULL DEFAULT NULL,
  `unique_rate_of_mapped` FLOAT NULL DEFAULT NULL,
  `dupli_rate_mapped` FLOAT NULL DEFAULT NULL,
  `dupli_rate_no_globins` FLOAT NULL DEFAULT NULL,
  `base_mismatch` FLOAT NULL DEFAULT NULL,
  `end_1_mapping_rate` FLOAT NULL DEFAULT NULL,
  `end_2_mapping_rate` FLOAT NULL DEFAULT NULL,
  `end_1_mismatch_rate` FLOAT NULL DEFAULT NULL,
  `end_2_mismatch_rate` FLOAT NULL DEFAULT NULL,
  `expression_profiling_efficiency` FLOAT NULL DEFAULT NULL,
  `high_quality_rate` FLOAT NULL DEFAULT NULL,
  `exonic_rate` FLOAT NULL DEFAULT NULL,
  `intronic_rate` FLOAT NULL DEFAULT NULL,
  `intergenic_rate` FLOAT NULL DEFAULT NULL,
  `intragenic_rate` FLOAT NULL DEFAULT NULL,
  `ambiguous_alignment_rate` FLOAT NULL DEFAULT NULL,
  `high_quality_exonic_rate` FLOAT NULL DEFAULT NULL,
  `high_quality_intronic_rate` FLOAT NULL DEFAULT NULL,
  `high_quality_intergenic_rate` FLOAT NULL DEFAULT NULL,
  `high_quality_intragenic_rate` FLOAT NULL DEFAULT NULL,
  `high_quality_ambiguous_alignment_rate` FLOAT NULL DEFAULT NULL,
  `discard_rate` FLOAT NULL DEFAULT NULL,
  `rrna_rate` FLOAT NULL DEFAULT NULL,
  `chimeric_alignment_rate` FLOAT NULL DEFAULT NULL,
  `end_1_sense_rate` FLOAT NULL DEFAULT NULL,
  `end_2_sense_rate` FLOAT NULL DEFAULT NULL,
  `avg_splits_per_read` FLOAT NULL DEFAULT NULL,
  `alter_alignments` FLOAT NULL DEFAULT NULL,
  `chimeric_reads` FLOAT NULL DEFAULT NULL,
  `duplicate_reads` FLOAT NULL DEFAULT NULL,
  `end_1_antisense` FLOAT NULL DEFAULT NULL,
  `end_2_antisense` FLOAT NULL DEFAULT NULL,
  `end_1_bases` FLOAT NULL DEFAULT NULL,
  `end_2_bases` FLOAT NULL DEFAULT NULL,
  `end_1_mapped_reads` FLOAT NULL DEFAULT NULL,
  `end_2_mapped_reads` FLOAT NULL DEFAULT NULL,
  `end_1_mismatches` FLOAT NULL DEFAULT NULL,
  `end_2_mismatches` FLOAT NULL DEFAULT NULL,
  `end_1_sense` FLOAT NULL DEFAULT NULL,
  `end_2_sense` FLOAT NULL DEFAULT NULL,
  `exonic_reads` FLOAT NULL DEFAULT NULL,
  `failed_vendor_qc` FLOAT NULL DEFAULT NULL,
  `high_quality_reads` FLOAT NULL DEFAULT NULL,
  `intergenic_reads` FLOAT NULL DEFAULT NULL,
  `intragenic_reads` FLOAT NULL DEFAULT NULL,
  `ambiguous_reads` FLOAT NULL DEFAULT NULL,
  `intronic_reads` FLOAT NULL DEFAULT NULL,
  `low_mapping_quality` FLOAT NULL DEFAULT NULL,
  `low_quality_reads` FLOAT NULL DEFAULT NULL,
  `mapped_duplicate_reads` FLOAT NULL DEFAULT NULL,
  `mapped_reads` FLOAT NULL DEFAULT NULL,
  `mapped_unique_reads` FLOAT NULL DEFAULT NULL,
  `mismatched_bases` FLOAT NULL DEFAULT NULL,
  `non_globin_reads` FLOAT NULL DEFAULT NULL,
  `non_globin_duplicate_reads` FLOAT NULL DEFAULT NULL,
  `reads_excluded_from_exon_counts` FLOAT NULL DEFAULT NULL,
  `reads_used_for_intron_exon_counts` FLOAT NULL DEFAULT NULL,
  `rrna_reads` FLOAT NULL DEFAULT NULL,
  `total_bases` FLOAT NULL DEFAULT NULL,
  `total_mapped_pairs` FLOAT NULL DEFAULT NULL,
  `total_reads` FLOAT NULL DEFAULT NULL,
  `unique_mapping_vendor_qc_passed_reads` FLOAT NULL DEFAULT NULL,
  `unpaired_reads` FLOAT NULL DEFAULT NULL,
  `read_length` FLOAT NULL DEFAULT NULL,
  `genes_detected` FLOAT NULL DEFAULT NULL,
  `estimated_library_complexity` FLOAT NULL DEFAULT NULL,
  `genes_used_in_3_bias` FLOAT NULL DEFAULT NULL,
  `mean_3_bias` FLOAT NULL DEFAULT NULL,
  `median_3_bias` FLOAT NULL DEFAULT NULL,
  `3_bias_std` FLOAT NULL DEFAULT NULL,
  `3_bias_mad_std` FLOAT NULL DEFAULT NULL,
  `3_bias_25th_percentile` FLOAT NULL DEFAULT NULL,
  `3_bias_75th_percentile` FLOAT NULL DEFAULT NULL,
  `average_fragment_length` FLOAT NULL DEFAULT NULL,
  `fragment_length_median` FLOAT NULL DEFAULT NULL,
  `fragment_length_std` FLOAT NULL DEFAULT NULL,
  `fragment_length_mad_std` FLOAT NULL DEFAULT NULL,
  `median_of_avg_transcript_coverage` FLOAT NULL DEFAULT NULL,
  `median_of_transcript_coverage_std` FLOAT NULL DEFAULT NULL,
  `median_of_transcript_coverage_cv` FLOAT NULL DEFAULT NULL,
  `median_exon_cv` FLOAT NULL DEFAULT NULL,
  `exon_cv_mad` FLOAT NULL DEFAULT NULL,
  `note` VARCHAR(64) NULL DEFAULT NULL,
  `chimeric_pairs` FLOAT NULL DEFAULT NULL,
  `num_gaps` FLOAT NULL DEFAULT NULL,
  `5_norm` FLOAT NULL DEFAULT NULL,
  `3_norm` FLOAT NULL DEFAULT NULL,
  `mean_per_base_cov` FLOAT NULL DEFAULT NULL,
  `fragment_length_stdDev` FLOAT NULL DEFAULT NULL,
  `estimated_library_size` FLOAT NULL DEFAULT NULL,
  `mapped` FLOAT NULL DEFAULT NULL,
  `total_purity_filtered_reads_sequenced` FLOAT NULL DEFAULT NULL,
  `rrna` FLOAT NULL DEFAULT NULL,
  `failed_vendor_qc_check` FLOAT NULL DEFAULT NULL,
  `mean_cv` FLOAT NULL DEFAULT NULL,
  `transcripts_detected` FLOAT NULL DEFAULT NULL,
  `mapped_paris` FLOAT NULL DEFAULT NULL,
  `cumul_gap_length` FLOAT NULL DEFAULT NULL,
  `gap_pct` FLOAT NULL DEFAULT NULL,
  `mapped_unique_rate_of_total` FLOAT NULL DEFAULT NULL,
  `mapped_unique` FLOAT NULL DEFAULT NULL,
  `alter_aligments` FLOAT NULL DEFAULT NULL,
  `fragment_length_mean` FLOAT NULL DEFAULT NULL,
  `split_reads` FLOAT NULL DEFAULT NULL,
  `base_mismatch_rate` FLOAT NULL DEFAULT NULL,
  `end_1_pct_sense` FLOAT NULL DEFAULT NULL,
  `num_covered_5` FLOAT NULL DEFAULT NULL,
  `duplication_rate_of_mapped` FLOAT NULL DEFAULT NULL,
  `end_2_pct_sense` FLOAT NULL DEFAULT NULL,
  PRIMARY KEY (`sample_id`))
ENGINE = InnoDB
DEFAULT CHARACTER SET = utf8;


DROP TABLE IF EXISTS `studies`;
CREATE TABLE `studies` (
  `id` int(3) NOT NULL AUTO_INCREMENT,
  `study` VARCHAR(64) NOT NULL,
  `pi` VARCHAR(64) NULL,
  `datatype` VARCHAR(32) NOT NULL,
  `center` VARCHAR(64) NOT NULL,
  `samplereceived` int(8) NULL,
  `updated_at` TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
  PRIMARY KEY (`id`),
  INDEX `idx_study_datatype` (`study` ASC, `datatype` ASC)
) ENGINE=InnoDB DEFAULT CHARSET=utf8;

INSERT INTO `studies` (`id`, `study`, `pi`, `datatype`, `center`, `samplereceived`) VALUES 
('1', 'GALAII', 'Burchard', 'RNA-seq', 'Broad', '2491'),
('2', 'SAGE', 'Burchard', 'RNA-seq', 'Broad', '950'),
('3', 'WHI', 'Kooperberg', 'RNA-seq', 'Broad', '1367'),
('4', 'Framingham', 'Levy', 'RNA-seq', 'NWGC', '1522'),
('5', 'Spiromics', 'Meyers', 'RNA-seq', 'NWGC', '3980'),
('6', 'Framingham', 'Ramachandran', 'RNA-seq', 'NWGC', '1191'),
('7', 'MESA pilot', 'Rotter', 'RNA-seq', 'Broad', '1500'),
('8', 'MESA pilot', 'Rotter', 'RNA-seq', 'NWGC', '1422'),
('9', 'PCGC', 'Seidman', 'RNA-seq', 'Broad', '95'),
('10', 'LTRC', 'Silverman', 'RNA-seq', 'NWGC', '2354'),
('11', 'TOP-CHeF', 'M.Taylor', 'RNA-seq', 'Broad', '654'),
('12', 'Sentinel', 'UW', 'RNA-seq', 'NWGC', '96'),
('13', 'PVDOMICS', 'Erzurum', 'Methylation', 'NWGC', '958'),
('14', 'CARDIA', 'Fornage', 'Methylation', 'USC', '1920'),
('15', 'WHI', 'Kooperberg', 'Methylation', 'USC', '1334'),
('16', 'Framingham', 'Ramachandran', 'Methylation', 'USC', '1814'),
('17', 'MESA pilot', 'Rotter', 'Methylation', 'USC', '2406'),
('18', 'COPDGene', 'Silverman', 'Methylation', 'NWGC', '11843'),
('19', 'LTRC', 'Silverman', 'Methylation', 'NWGC', '3041'),
('20', 'CAMP', 'Weiss', 'Methylation', 'USC', '1616'),
('21', 'CRA', 'Weiss', 'Methylation', 'USC', '1238'),
('22', 'WHI', 'Kooperberg', 'Metabolomics', 'Broad', '1400'),
('23', 'Framingham', 'Ramachandran', 'Metabolomics', 'Broad', '3025'),
('24', 'CAMP', 'Weiss', 'Metabolomics', 'Broad', '962'),
('25', 'CRA', 'Weiss', 'Metabolomics', 'Broad', '2038');
